public/app/features/chat/Chat.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { PureComponent } from 'react';
+import { TextArea } from '@grafana/ui';
+import { getGrafanaLiveSrv, getBackendSrv } from '@grafana/runtime';
+import { isLiveChannelMessageEvent, LiveChannelScope } from '@grafana/data';
+import { Unsubscribable } from 'rxjs';
+import { ChatMessageAction } from './ChatMessageAction';
+import { Message, ChatMessage } from './ChatMessage';
+
+interface MessagePacket {
+  event: string;
+  messageCreated: Message;
+}
+
+export interface ChatProps {
+  contentTypeId: number;
+  objectId: string;
+  actions?: ChatMessageAction[];
+}
+
+export interface ChatState {
+  messages: Message[];
+  value: string;
+}
+
+export class Chat extends PureComponent<ChatProps, ChatState> {
+  subscription?: Unsubscribable;
+  chatInput?: any;
+  chatContainer?: any;
+
+  state: ChatState = {
+    messages: [],
+    value: '',
+  };
+
+  constructor(props: ChatProps) {
+    super(props);
+  }
+
+  async componentDidMount() {
+    const resp = await getBackendSrv().post('/api/chats/get-messages/', {
+      objectId: this.props.objectId,
+      contentTypeId: this.props.contentTypeId,
+    });
+    this.setState({
+      messages: resp.chatMessages,
+    });
+    this.updateSubscription();
+    this.scrollToBottom();
+    this.focusInput();
+  }
+
+  handleChange = (e: any) => {
+    this.setState({ value: e.target.value });
+  };
+
+  sendMessage = async () => {
+    if (!this.state.value) {
+      return;
+    }
+    await getBackendSrv().post('/api/chats/send-message/', {
+      objectId: this.props.objectId,
+      contentTypeId: this.props.contentTypeId,
+      content: this.state.value,
+    });
+    this.setState({
+      value: '',
+    });
+  };
+
+  getLiveAddr = () => {
+    return {
+      scope: LiveChannelScope.Grafana,
+      namespace: 'chat', // holds on to the last value
+      path: `${this.props.contentTypeId}/${this.props.objectId}`,
+    };
+  };
+
+  getLiveChannel = () => {
+    const live = getGrafanaLiveSrv();
+    if (!live) {
+      console.error('Grafana live not running, enable "live" feature toggle');
+      return undefined;
+    }
+
+    const addr = this.getLiveAddr();
+    if (!addr) {
+      return undefined;
+    }
+    return live.getStream<MessagePacket>(addr);
+  };
+
+  onKeyboardPress = async (e: any) => {
+    if (e.key === 'Enter' && e.shiftKey) {
+      e.preventDefault();
+      let start = e.target.selectionStart,
+        end = e.target.selectionEnd;
+      this.setState(
+        (prevState) => ({ value: prevState.value.substring(0, start) + '\n' + prevState.value.substring(end) }),
+        () => {
+          this.chatInput.selectionStart = this.chatInput.selectionEnd = start + 1;
+        }
+      );
+    } else if (e.key === 'Enter') {
+      e.preventDefault();
+    }
+  };
```

Comment:
Not sure what this is does with the selection? 

Nitpick: do not think enter key should submit the message. Enter is useful for newlines and formatting the message. A button is likely better. 
![Uploading Screenshot from 2022-02-09 20-51-16.png…]()

Reply:
Just updated / cleaned up frontend, did not include button at this time, however user can make newlines via "shift" + "enter".

On a whole I think that "enter" to submit the comment is inline with UX paradigms (similar to Slack's UX). As we refine this I think that we may want to include a separate send button eventually (for accessibility sake alone) as well as ability to edit / delete comments etc.